File: docs/snippets/all/archetypes/mcap_schema_simple.py
"""Log a simple MCAP schema definition."""

import dalaran as dl

dl.init("dalaran_example_mcap_schema", spawn=True)

# Example ROS2 message definition for a simple Point message
point_schema = """float64 x
float64 y
float64 z"""

dl.log(
    "mcap/schemas/geometry_point",
    dl.McapSchema(
        id=42,
        name="geometry_msgs/msg/Point",
        encoding="ros2msg",
        data=point_schema.encode("utf-8"),
    ),
)

File: docs/snippets/all/archetypes/mcap_schema_simple.rs
//! Log a simple MCAP schema definition.

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let rec = dalaran::RecordingStreamBuilder::new("dalaran_example_mcap_schema")
        .spawn()?;

    // Example ROS2 message definition for a simple Point message
    let point_schema = "float64 x\nfloat64 y\nfloat64 z";

    rec.log(
        "mcap/schemas/geometry_point",
        &dalaran::McapSchema::new(
            42,
            "geometry_msgs/msg/Point",
            "ros2msg",
            point_schema.as_bytes(),
        ),
    )?;

    Ok(())
}
